aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2512,6 +2553,21 @@ pub(crate) fn is_account_init_for_sponsored_transaction(
     )
 }
 
+fn check_randomness_annotation(
+    session: &mut SessionExt,
+    resolver: &impl AptosMoveResolver,
+    payload: &TransactionPayload,
+) -> bool {
+    match payload {
+        TransactionPayload::Script(_)
+        | TransactionPayload::ModuleBundle(_)
+        | TransactionPayload::Multisig(_) => false,
+        TransactionPayload::EntryFunction(entry_func) => {
+            has_randomness_attribute(resolver, session, entry_func).unwrap_or(false)
```

Comment:
This should propagate the error, which can happen if module loading failed

Reply:
(i'm turning this func into `get_required_deposit`)
can this function focus on required deposit calculation and let a later step capture the module loading failure?